### PR TITLE
server: fix REST connection

### PR DESCRIPTION
**Pull request by guggero**
Because of a misplaced defer of a context cancel, the REST proxy didn't
work at all before the fix in this commit.

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -254,9 +255,6 @@ func (s *Server) Start() error {
 
 		log.Infof("RPC server listening on %s", s.grpcListener.Addr())
 		if s.restListener != nil {
-			s.restListener = tls.NewListener(
```
Comment (halseth) [inline server.go:257]: hm, why was this here?

Comment (guggero) [reply server.go:257]: I wondered about this myself... Should I remove it in a separate commit to make it more clear this was wrong in the first place?

Comment (halseth) [reply server.go:257]: Yes 👍

Comment (guggero) [reply server.go:257]: Done.

File: server.go [hunk 2]
```diff
@@ -194,8 +195,8 @@ func (s *Server) Start() error {
 
 		// We'll also create and start an accompanying proxy to serve
 		// clients through REST.
-		ctx, cancel := context.WithCancel(context.Background())
-		defer cancel()
```
Comment (Roasbeef) [inline server.go:198]: Nice catch!

**Review by Roasbeef — APPROVED** [review]
LGTM 🐋

**Review by halseth — APPROVED** [review]
LGTM 🕺

**Comment by halseth** [comment]
@guggero Linter failing. Can merge once Travis pass.

File: server.go [hunk 3]
```diff
@@ -194,8 +195,8 @@ func (s *Server) Start() error {
...
+		ctx := context.Background()
```
Comment (halseth) [inline server.go:198]: can remove this
